components/style/values/specified/font.rs

```diff
@@ -610,6 +611,296 @@ impl Parse for FontSize {
+    Value(VariantAlternatesList),
+    /// Use system font glyph
+    System(SystemFont)
+}
+
+impl FontVariantAlternates {
+    #[inline]
+    /// Get initial specified value with VariantAlternatesList
+    pub fn get_initial_specified_value() -> Self {
+        FontVariantAlternates::Value(VariantAlternatesList(vec![].into_boxed_slice()))
+    }
+
+    /// Get FontVariantAlternates with system font
+    pub fn system_font(f: SystemFont) -> Self {
+        FontVariantAlternates::System(f)
+    }
+
+    /// Get SystemFont of FontVariantAlternates
+    pub fn get_system(&self) -> Option<SystemFont> {
+        if let FontVariantAlternates::System(s) = *self {
+            Some(s)
+        } else {
+            None
+        }
+    }
+}
+
+impl ToComputedValue for FontVariantAlternates {
+    type ComputedValue = computed::FontVariantAlternates;
+
+    fn to_computed_value(&self, _context: &Context) -> computed::FontVariantAlternates {
+        match *self {
+            FontVariantAlternates::Value(ref v) => v.clone(),
+            FontVariantAlternates::System(_) => {
+                #[cfg(feature = "gecko")] {
+                    _context.cached_system_font.as_ref().unwrap().font_variant_alternates.clone()
+                }
+                #[cfg(feature = "servo")] {
+                    unreachable!()
+                }
+            }
+        }
+    }
+
+    fn from_computed_value(other: &computed::FontVariantAlternates) -> Self {
+        FontVariantAlternates::Value(other.clone())
+    }
+}
+
+impl Parse for FontVariantAlternates {
+    /// normal |
+    ///  [ stylistic(<feature-value-name>)           ||
+    ///    historical-forms                          ||
+    ///    styleset(<feature-value-name> #)          ||
+    ///    character-variant(<feature-value-name> #) ||
+    ///    swash(<feature-value-name>)               ||
+    ///    ornaments(<feature-value-name>)           ||
+    ///    annotation(<feature-value-name>) ]
+    fn parse<'i, 't>(_: &ParserContext, input: &mut Parser<'i, 't>) -> Result<FontVariantAlternates, ParseError<'i>> {
+        let mut alternates = Vec::new();
+        if input.try(|input| input.expect_ident_matching("normal")).is_ok() {
+            return Ok(FontVariantAlternates::Value(VariantAlternatesList(alternates.into_boxed_slice())));
+        }
+
+        let mut parsed_alternates = ParsingFlags::empty();
+        macro_rules! check_if_parsed(
+            ($input:expr, $flag:path) => (
+                if parsed_alternates.contains($flag) {
+                    return Err($input.new_custom_error(StyleParseErrorKind::UnspecifiedError))
+                }
+                parsed_alternates |= $flag;
+            )
+        );
+        while let Ok(_) = input.try(|input| {
+            // FIXME: remove clone() when lifetimes are non-lexical
+            match input.next()?.clone() {
+                Token::Ident(ref ident) => {
+                    if *ident == "historical-forms" {
```

Comment:
This should be `ident.eq_ignore_ascii_case("historical-forms")`, could you add a test for this?

Something in the lines of:

```html
<!doctype html>
<div style="font-variant-alternates: Historical-Forms"></div>
<script>
getComputedStyle(document.querySelector('div')).fontVariantAlternates == "historical-forms"
</script>
```

Though... I guess it would need to be a test for Firefox, not servo, so let's fix it and I'll land the test in https://bugzilla.mozilla.org/show_bug.cgi?id=1415946